types/wavesurfer.js/src/mediaelement.d.ts

```diff
@@ -0,0 +1,49 @@
+import { Peaks, WebAudioBackend } from "../types/backend";
+import { WaveSurferParams } from "../types/params";
+
+export default class MediaElement extends WebAudioBackend {
+    constructor(params: WaveSurferParams);
+
+    /** Create a timer to provide a more precise audioprocess event. */
+    createTimer(): void;
+    /** This is called when wavesurfer is destroyed. */
+    destroy(): void;
+    /** Returns the current time in seconds relative to the audio-clip's duration. */
+    getCurrentTime(): number;
+    /** Used by wavesurfer.getDuration(). */
+    getDuration(): number;
+    /** Compute the max and min value of the waveform when broken into <length> subranges. */
+    getPeaks(length: number, first: number, last: number): Peaks;
+    /** Get the audio source playback rate. */
+    getPlaybackRate(): number;
+    /** Get the position from 0 to 1. */
+    getPlayedPercents(): number;
+    /** Get the current volume. */
+    getVolume(): number;
+    /** Initialise the backend, called in wavesurfer.createBackend(). */
+    init(): void;
+    /** Used by wavesurfer.isPlaying() and wavesurfer.playPause(). */
+    isPaused(): boolean;
+    /** Create media element with url as its source, and append to container element. */
+    load(url: string, container: HTMLElement, peaks: Peaks, preload: string): void;
+    /** Load existing media element. */
+    loadElt(elt: HTMLMediaElement, peaks: Peaks): void;
+    /** Pauses the loaded audio. */
+    pause(): Promise<void>;
```

Comment:
This isn't your fault, but the WaveSurfer jsdoc comments are wrong and this method actually returns nothing, not a Promise.

https://github.com/katspaugh/wavesurfer.js/blob/11f31aeeab7f7d4e244530a62b4e973e8509dac1/src/mediaelement.js#L307

HTMLMediaElement.pause() returns nothing: https://developer.mozilla.org/en-US/docs/Web/API/HTMLMediaElement/pause


edit: I guess it really returns `void | undefined`? :S

Reply:
I think in type declarations, `undefined` is basically the same as `void`. I guess `void` is just `undefined` in the HTML Standard, but the vscode tooltip shows `void`:

![grafik](https://user-images.githubusercontent.com/495429/117497252-8b200580-af78-11eb-972e-ea454df9dfb0.png)

Reply:
Does this also affect `WaveSurfer.pause()`?

PS: I will just remove `MediaElement.pause(): void`, because it is already inherited from the (helper type) `WebAudioBackend`.

Reply:
> Does this also apply to WaveSurfer.pause()?

It does, actually.

Reply:
Fixed in aa40ff5.

Reply:
sorry I never got emails about these replies but it looks like you figured it out :)